### PR TITLE
Feat(Hypernative): Add guard detection service and hook

### PR DESCRIPTION
**Pull request by katspaugh**
## Summary

Resolves [COR-772](https://linear.app/safe-global/issue/COR-772/distinguish-hypernative-from-non-hypernative-users)

Implement functionality to detect if a Safe has a HypernativeGuard installed by comparing the guard contract's bytecode hash against known HypernativeGuard deployments.

## Features
- ✅ Service layer with `isHypernativeGuard()` function to check guard contracts
- ✅ React hook `useIsHypernativeGuard()` that returns `{ isHypernativeGuard, loading }`
- ✅ Comprehensive test coverage (18 tests, all passing)
- ✅ Utility script to fetch and hash contract bytecode
- ✅ Example components showing usage patterns
- ✅ Full documentation in README.md

## How It Works
1. Hook checks if the Safe has a `guard` field set via `useSafeInfo().safe.guard`
2. If guard exists, fetches the contract bytecode at that address using `web3ReadOnly.getCode()`
3. Hashes the bytecode using `keccak256`
4. Compares the hash against known HypernativeGuard deployment hashes for that chain
5. Returns `true` if match found, `false` otherwise

## Usage Example
```typescript
import { useIsHypernativeGuard } from '@/features/hypernative/hooks'

function MyComponent() {
  const { isHypernativeGuard, loading } = useIsHypernativeGuard()

  if (loading) return <div>Checking...</div>
  
  return isHypernativeGuard ? (
    <div>Protected by HypernativeGuard ✓</div>
  ) : (
    <div>No HypernativeGuard detected</div>
  )
}
```

## Files Added
- `src/features/hypernative/services/hypernativeGuardCheck.ts` - Core service logic
- `src/features/hypernative/services/__tests__/hypernativeGuardCheck.test.ts` - Service tests (10 tests)
- `src/features/hypernative/hooks/useIsHypernativeGuard.ts` - React hook wrapper
- `src/features/hypernative/hooks/__tests__/useIsHypernativeGuard.test.ts` - Hook tests (8 tests)
- `src/features/hypernative/hooks/index.ts` - Export index
- `src/features/hypernative/scripts/fetchGuardCodeHash.ts` - Utility script
- `src/features/hypernative/examples/HypernativeGuardBadge.tsx` - Example components
- `src/features/hypernative/README.md` - Complete documentation

## Test Plan
- [x] All unit tests pass (18/18)
- [x] Service correctly identifies HypernativeGuard contracts by code hash
- [x] Hook handles all edge cases (no guard, no provider, loading states, errors)
- [x] Hook re-checks when guard address or chain ID changes
- [x] TypeScript types are correct

## Next Steps
To use this in production, you need to:
1. Fetch the actual Sepolia HypernativeGuard bytecode hash from `0x4784e9bF408F649D04A0a3294e87B0c74C5A3020`
2. Add the hash to `HYPERNATIVE_GUARD_CODE_HASHES` in `hypernativeGuardCheck.ts`
3. Add hashes for other networks as HypernativeGuard gets deployed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> Adds a service and React hook to detect HypernativeGuard via bytecode hash, with tests and documentation.
> 
> - **Services**:
>   - `services/hypernativeGuardCheck.ts`: Implements `isHypernativeGuard()` comparing guard bytecode hash to `HYPERNATIVE_GUARD_CODE_HASHES`; logs errors via `logError(Errors._809)`.
> - **Hooks**:
>   - `hooks/useIsHypernativeGuard.ts`: Exposes `{ isHypernativeGuard, loading }`, integrates `useSafeInfo`, `useWeb3ReadOnly`, and `useAsync`.
>   - `hooks/index.ts`: Exports hook and types.
> - **Tests**:
>   - `services/__tests__/hypernativeGuardCheck.test.ts`: Covers hash matching, empty code, no hashes, provider errors, multiple hashes.
>   - `hooks/__tests__/useIsHypernativeGuard.test.ts`: Covers loading states, no guard/provider, positive/negative detection, errors, re-checks, race conditions, resets.
> - **Docs**:
>   - `README.md`: Usage examples, architecture, testing instructions, and guidance for adding new code hashes.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 64b14dbbb30c7bcf2b4d559be701804e00cdd5c5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by katspaugh** [comment]
@claude review please.

**Comment by katspaugh** [comment]
The failing checks are from the parent branch.

File: apps/web/src/features/hypernative/services/hypernativeGuardCheck.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { keccak256 } from 'ethers'
+import type { JsonRpcProvider } from 'ethers'
+import { logError, Errors } from '@/services/exceptions'
+
+/**
+ * Known code hashes for HypernativeGuard contracts
+ * HypernativeGuard has the same bytecode across all chains
+ *
+ * Deployment example: Sepolia at 0x4784e9bF408F649D04A0a3294e87B0c74C5A3020
```
Comment (vmosharova) [inline apps/web/src/features/hypernative/services/hypernativeGuardCheck.ts]: Does it make sense to test it on other networks as well?
